app/src/main/java/com/github/se/icebreakrr/ui/authentication/SignIn.kt

```diff
@@ -0,0 +1,235 @@
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.icebreakrr.R
+import com.github.se.icebreakrr.config.LocalIsTesting
+import com.github.se.icebreakrr.ui.navigation.NavigationActions
+import com.github.se.icebreakrr.ui.navigation.TopLevelDestinations
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.firebase.Firebase
+import com.google.firebase.auth.AuthResult
+import com.google.firebase.auth.GoogleAuthProvider
+import com.google.firebase.auth.auth
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+
+/**
+ * Composable function that displays the SignIn screen. This screen handles the user authentication
+ * flow with Google Sign-In via Firebase.
+ *
+ * @param navigationActions A class that handles the navigation between screens.
+ */
+@Composable
+fun SignInScreen(navigationActions: NavigationActions) {
+
+  // State to hold the current Firebase user
+  var user by remember { mutableStateOf(Firebase.auth.currentUser) }
+  // Retrieve the token from resources (needed for Google Sign-In)
+  val token = stringResource(R.string.default_web_client_id)
+  // Get the context to use within the composable
+  val context = LocalContext.current
+
+  // Check if the app is running in test mode
+  val isTesting = LocalIsTesting.current
+
+  // Get screen configuration to dynamically adjust layout
+  val configuration = LocalConfiguration.current
+  val screenHeight = configuration.screenHeightDp
+  val screenWidth = configuration.screenWidthDp
+
+  // Set padding dynamically based on screen size
+  val verticalPadding =
+      when {
+        screenHeight > 800 -> 150.dp
+        else -> 100.dp
+      }
+
+  val verticalSpacing =
+      when {
+        screenHeight > 800 -> 250.dp
+        else -> 150.dp
+      }
```

Comment:
fun fact: in kotlin you can do both this assignement in a single `when` block with the help of the `to` keyword : 

```
val (verticalPadding, verticalSpacing) = when {
    screenHeight > 800 -> 150.dp to 250.dp
    else -> 100.dp to 150.dp
```
see https://kotlinlang.org/docs/destructuring-declarations.html

Reply:
Thank you for the tips!

Reply:
Wow thanks for the tip ! :)